src/common/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import styles from './Dropdown.module.css';
+
+interface DropdownOption {
+  id: string;
+  label: string;
+  icon?: string;
+  url?: string;
+  onClick?: () => void;
+}
+type DropdownOptions = DropdownOption[];
+
+type Props = {
+  options: DropdownOptions;
+};
+
+function Dropdown(props: Props) {
+  const [isOpen, toggleDropdown] = useState(false);
+  const { t } = useTranslation();
+
+  const isDropdown = props.options.length > 1;
+  const navBarItem = props.options[0];
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  const handleClick = (event: Event) => {
+    if (ref.current && !ref.current.contains(event.target as Node)) {
+      toggleDropdown(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick);
+    return () => {
+      document.removeEventListener('click', handleClick);
+    };
+  }, []);
+
+  return (
+    <div className={styles.dropdownWrapper} ref={ref}>
+      {/* The button text is the first item in the option list */}
+      {isDropdown && (
+        <React.Fragment>
+          <button
+            className={styles.navButton}
+            onClick={() => toggleDropdown(prevState => !prevState)}
+          >
+            {navBarItem.label}
+            <img
+              src={navBarItem.icon}
+              alt={t('nav.menuButtonLabel')}
```

Comment:
Consider whether you really need the `alt` attribute here. When the image is right next to a text that describes its purpose, you don't necessarily need to label the image separately. You do need the alt text if the icon is more than just decorative. If you use this element with a screen reader it would read out `navBarItem.label` and then `t('nav.menuButtonLabel')` right after. Would that be valuable?

To avoid linting errors, you must use `alt=""`.

> This means that the alt attribute (sometimes called the alt tag, though technically this is incorrect) is not the only mechanism for providing the content and function of the image. This information can also be provided in text adjacent to the image or within the page containing the image. In some cases where the equivalent cannot be presented succinctly, a link to a separate page that contains a longer description of the image content can be provided.

https://webaim.org/techniques/alttext/

If the icon is more than just decorative, consider whether the alt text should be changed from menu to something else.

Reply:
I tested the menu out with voice over, and currently it reads (when logged in).

"FirstName LastName Menu"

I think the icon does have more than just a decorative purpose here. It denotes--to the seeing user--that this is a profile menu.

I think that the most correct way to resolve the issue would be to change the alt text for the icon to be "profile menu". This would meant that you should probably pass it as a prop for the `dropdown` component instead of declaring it within the component itself.

Reply:
Alt text is now passed to dropdown and its value is changed to "Profile menu".